addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ParamSinksUtils.java

```diff
@@ -188,15 +156,48 @@ private static String getCachedItem(Map<String, String> map, String item) {
 
     private static class UserDataSource {
 
+        private static final String GENERIC_STRING = "DYX";
         private final String uri;
         private final String param;
         private final String stringRepresentation;
 
         public UserDataSource(HttpMessage sourceMsg, String param) {
             super();
-            this.uri = getCachedItem(cachedUris, sourceMsg.getRequestHeader().getURI().toString());
-            this.param = getCachedItem(cachedParams, param);
-            this.stringRepresentation = uri + "#" + param;
+
+            String genericUri = createGenericUri(sourceMsg.getRequestHeader().getURI(), param);
+            this.uri = getCachedItem(cachedUris, genericUri);
+
+            boolean requestUriIsGeneric = sourceMsg.getRequestHeader().getURI().equals(genericUri);
+            if (!requestUriIsGeneric) {
```

Comment:
Name is giving wrong indication like if they are equal then it is not generic but our flag isGeneric become true. may be a `!` ahead of in 170 line can be more correct.

Reply:
Yes the name is not well chosen, I will change it.